docs/zkapps/tutorials/04-zkapp-ui-with-react.mdx

```diff
@@ -161,15 +170,143 @@ Because o1js code is computationally intensive, it's helpful to use web workers.
 
 ### Download the main browser UI logic file
 
-The example project has a completed app. The `index.page.tsx` file is the entry file for your application and contains the main logic for the browser UI that is ready to deploy to GitHub Pages.
+The example project has a completed app. The `page.tsx` file is the entry file for your application and contains the main logic for the browser UI that is ready to deploy to GitHub Pages.
+
+1. Download the [page.tsx](https://github.com/o1-labs/docs2/blob/main/examples/zkapps/04-zkapp-browser-ui/ui/app/page.tsx) example file.
+
+1. Move the `page.tsx` file to your local `04-zkapp-browser-ui/ui/app` directory.
+
+### Setup web workers
+
+The web worker code resides in the `04-zkapp-browser-ui/ui/app/zkappWorker.ts` file. Here, you define the functions that will be executed in the worker thread.
+
+#### Defining State
+
+The state object holds references to the zkApp Instance, Add contract instance, and the transactions.
+```ts ignore
+  const state = {
+    AddInstance: null as null | typeof Add,
+    zkappInstance: null as null | Add,
+    transaction: null as null | Transaction,
+  };
+  ```
+
+#### Functions using `o1js` that will be executed in the worker thread
```

Comment:
maybe something like this would be easier to understand?

```
##### Defining functions that will run in the worker thread

Brief explanation of what the code snippet does
```